nncf/common/sparsity/schedulers.py

```diff
@@ -11,207 +11,289 @@
  limitations under the License.
 """
 
-import numpy as np
-from bisect import bisect_right
+from typing import Optional, Dict
 
 from nncf.common.utils.logger import logger
 from nncf.common.utils.registry import Registry
+from nncf.common.schedulers import PolynomialDecaySchedule
+from nncf.common.schedulers import ExponentialDecaySchedule
+from nncf.common.schedulers import MultiStepSchedule
 from nncf.api.compression import CompressionLevel
 from nncf.api.compression import CompressionScheduler
 
 SPARSITY_SCHEDULERS = Registry("sparsity_schedulers")
 
 
 class SparsityScheduler(CompressionScheduler):
-    def __init__(self, controller, params: dict = None):
+    """
+    This is the class from which all sparsity schedulers inherit.
+
+    A sparsity scheduler is an object which specifies the sparsity
+    level at each training epoch or each training step. It involves a
+    scheduling algorithm, defined in the `_calculate_sparsity_level()`
+    method and a state (some parameters required for current sparsity
+    level calculation) defined in the `__init__()` method.
+    """
+
+    def __init__(self, controller, params: dict):
+        """
+        Initializes the internal state of the sparsity scheduler specified by:
+            - controller: Sparsity algorithm controller.
+            - initial_sparsity: Sparsity level which already has been
+                applied to the model. It is the level at which the schedule begins.
+            - target_sparsity: Sparsity level at which the schedule ends.
+            - target_epoch: Zero-based index of the epoch from which the
+                sparsity level of the model will be equal to the `target_sparsity`.
+            - freeze_epoch: Zero-based index of the epoch from which the sparsity
+                mask will be frozen and will not be trained.
+            - current_sparsity: Sparsity level for the `current_epoch`.
+
+        :param controller: Sparsity algorithm controller.
+        :param params: Parameters of the scheduler.
+        """
         super().__init__()
-        if params is None:
-            self._params = dict()
-        else:
-            self._params = params
-
         self.controller = controller
         self.initial_sparsity = self.controller.get_sparsity_init()
-        self.target_sparsity = self._params.get('sparsity_target', 0.5)
-        self.target_epoch = self._params.get('sparsity_target_epoch', 90)
-        self.freeze_epoch = self._params.get('sparsity_freeze_epoch', 100)
-
-    def _set_sparsity_level(self):
+        self.target_sparsity = params.get('sparsity_target', 0.5)
+        self.target_epoch = params.get('sparsity_target_epoch', 90)
+        self.freeze_epoch = params.get('sparsity_freeze_epoch', 100)
```

Comment:
@vshampor @alexsu52 What are these default values for? Can we make these parameters required?

Reply:
My opinion, `sparsity_target` and `sparsity_target_epoch` is required parameters and `sparsity_freeze_epoch` is optional parameter.

Reply:
Let's discuss it in another PR